src/Model.jl

```diff
@@ -21,8 +21,14 @@ struct Model{T <: Real, VT <: Real}
     unit_cell_volume::T
     recip_cell_volume::T
 
-    # Electrons, occupation and smearing function
-    n_electrons::Int  # usually consistent with `atoms` field, but doesn't have to
+    # If not `nothing` then do computations in the canonical ensemble, where
+    # `n_electrons` is fixed; electron count usually consistent with `atoms` field.
+    n_electrons::Union{Int, Nothing}
+    # Expert option: If not `nothing` then do computations in the grand-canonical ensemble,
+    # where `εF` is fixed. Computations with Coulomb electrostatics will fail by default.
+    # Use `check_electrostatics=false` to disable these checks.
+    εF::Union{T, Nothing}
+    check_electrostatics::Bool  # Enables checks that electrostatics is consistent
```

Comment:
This is a debug option, I really don't like having it as a field in model. Just remove the check_electrostatics and have a warning?